### ISSUE TITLE
getting the climate data for Team Background

### ISSUE THREAD

**Issue by lizzieinvancouver** [AileneKane/radcliffe#42]
Current email train!

Me to Ben Cook on March 20 2017:

I offered to take this task back over from Ailene at her request. And by 'this' I mean sorting out the climate data summarized for each gridcell in a species range and for the sites we have experimental data for -- I checked with Christy and the latter is done but not the former. I believe you did each site here:

https://github.com/AileneKane/radcliffe/commit/72845dc5ae3a6ff4ddd324ff8a2bcfa00850ce99

What we need now is the climate variables for each GRIDCELL of a species range. Can you help update the code to also return files for each species?

Semi-relatedly, I checked with Christy on whether GLDAS might get updated soon... if it won't I can help try to figure out what species we don't need data for based on which sites we don't have GLDAS data for. 

On 3/14/17 5:14 PM, Christy Rollinson wrote:
> Hi Lizzie,
>
> I was actually about to email you (about something different, but still phenology-related).  First: yes, I did extract met for each grid cell of the species’ range.  Those are stored on [Cyverse](https://de.cyverse.org/de/), which you’ll need to register for (it’s free) and then I can give you access to the [folder](https://de.cyverse.org/de/?type=data&folder=/iplant/home/crollinson/radcliffe/SpeciesMet).
>
 From Ailene on March 1:

I am ready to work on my task, which is to calculate "the climate in (a) the southern range (measured as the warmest quartile of cells in the range, using the 4 temp variables we have) and (b) the core climate (measured at the centroid (using the 4 temp variables we have) or …?)” according to the wiki! <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_AileneKane_radcliffe_wiki_Team-2Dbackground-3A-2DNotes-2Dfrom-2D6-2DApril-2D2016-2D-28updated-2D16-2DDec-2D2016-29&d=CwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=555-bAufKEAVxrAD00rrvVjkskZqyAPlDlGpHbP_bOg&m=x8r8RnIImrpW-wElh1hoFFIyeRvD9upr9JiUKF3bjKo&s=ia-KzhJKa9C-oHRDkYvSx_iTktKE1LU0CjoQgaa7nJI&e=>

Thanks to all your great work on this so far, we already have a data file for each climate variable at each experimental study site. the data file has a summary of the focal climate variable for each month.
In addition to these, what we WANT (I think!) is the same climate variables for each grid cell in each species’ home range. I didn’t see this in the git repo.

I think that the climate variables have not been calculated yet for the species ranges. Is this true, Christy? If it is NOT true, please let me know where I can find these, and I will use them to summarize the focal climate variables for the southern range and the core climate of each species.

If it IS true, could one/both of you get the climate data for the species’ range? As I understand it, Christy has the range data for each species and the corresponding climate data, but Ben has been doing the climate calculations to date. I’m not if it is easier for Ben to do this, after Christy shares the data with him, or for Christy to do this after Ben explains his code for calculating the climate variables.Perhaps you two have a better sense of this?

**Comment by lizzieinvancouver** [comment]
Whoop! We're getting close. Here's an email from Ben on 22 March 2017:

Hi Lizzie,

Okay, I’m starting to get organized for this. So, just to confirm, for each of the species, and each gridcell in that species range, I am going to calculate:
        mean temperature of the coldest month (C)
        mean temperature of the warmest month (C)
        number of chilling days (1 Sept - 31 Dec) (days where mean temperature is less than 5 degrees C)
        Cumulative GDD, base 0, Jan 1 - Mar 31
	Cumulative GDD, base 0, Jan 1 - May 31    
        Cumulative GDD, base 5, Jan 1 - Mar 31
	Cumulative GDD, base 5, Jan 1 - May 31       
	total amount of precipitation: April-Sept (mm)
        total amount of precipitation: Oct-March (mm)

These are the same quantities I calculated for each of the experimental and observational sites. Because this particular analysis is supposed to focus on space (not time), I will provide long term (1949-2010) averages for all gridcells in the species range.

I’m going to get started on this tomorrow and I’ll check in when I have the first set of files generated, but feel free to give me any comments or questions when you like!

~Ben

**Comment by lizzieinvancouver** [comment]
And see my commit cdf8607be7bbec070b90fdf316468a2bf1a2ed4b for a check of the first file he sent. Looks good to me.

**Comment by lizzieinvancouver** [comment]
Onward! Email from Christy on 4 Apr 2017 (we were discussing some missing data Ben came across and how to fix it):

Hi Lizzie,

Sorry this fell off my radar — I could’ve sworn this email was last week, not 2 weeks ago!  I just looked through my scripts and the state of GLDAS and here’s what I found.  
1.) Luckily I saved all GLDAS 2.0 data for North America where we have species ranges locally, so I can re-extract species once I check what happened. So I can regenerate what I had.
2.) [GLDAS 2.1](https://hydro1.gesdisc.eosdis.nasa.gov/opendap/GLDAS/GLDAS_NOAH025_3H.2.1/) runs from 2000-present.  I haven’t checked to see if this is behind the authentication wall yet, but if not, I can easily extract the met for all sites (globally) for 2000-present.  If it is behind an authentication wall, there’s not too much data that it would be forever to download (I think)

The other alternative is to switch met products all together.  There are a number of global and US-centric products that will give us daily data, but only go back as far as 1980. Some options
-- [Daymet](https://daymet.ornl.gov/) includes US, Canada and Mexico, which is good for the tree species for which we have ranges.  For better or worse though, Daymet is very high resolution, so our file sizes are going to explode.  
      Pros: I know I have working scripts for getting Daymet data down & formatted; native daily temporal resolution; Daymet is a good standard dataset that I think a lot of ecologists in the US are familiar with; 
      Cons: Large datasets, will be slow, no Europe, earliest year = 1980ish
— [MERRA-2](https://gmao.gsfc.nasa.gov/reanalysis/MERRA-2/data_access/) is a re-analysis product (I think) and I know some people that use it for ecosystem modeling; 
      Pros: Europe available, native daily data available (I think), goes through 2017, download *should* work, 
      Cons: coarser resolution (0.5-degree); I haven’t worked with this dataset before (although I have scripts that should work); may also have been moved behind authentication wall.

Let me know what you think and how you want to proceed.  Sorry this has become such a pain in the butt — every time I turn around things have changed with met products!

Christy
